File: app/src/main/java/br/com/algorit/mangaalert/util/Notification.java
package br.com.algorit.mangaalert.util;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.AsyncTask;
import android.os.Build;

import java.lang.ref.WeakReference;

public class Notification {

    public Notification(Context context) {
        WeakReference<Context> weakContext = new WeakReference<>(context);
        new NotificationAsyncTask(weakContext).execute();
    }


    private static class NotificationAsyncTask extends AsyncTask<Void, Void, Void> {

        private final WeakReference<Context> weakContext;

        NotificationAsyncTask(WeakReference<Context> weakContext) {
            this.weakContext = weakContext;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                CharSequence name = "MangaChannel";
                String description = "Channel for Manga Reminder";
                NotificationChannel channel = new NotificationChannel("manga", name, NotificationManager.IMPORTANCE_DEFAULT);
                channel.setDescription(description);

                NotificationManager notificationManager = weakContext.get().getSystemService(NotificationManager.class);
                notificationManager.createNotificationChannel(channel);
            }

            return null;
        }
    }
}

File: app/src/main/java/br/com/algorit/mangaalert/util/Notification.kt
package br.com.algorit.mangaalert.util

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.AsyncTask
import android.os.Build
import java.lang.ref.WeakReference

class Notification(context: Context) {

    init {
        val weakContext = WeakReference(context)
        NotificationAsyncTask(weakContext).execute()
    }

    private class NotificationAsyncTask internal constructor(private val weakContext: WeakReference<Context>) : AsyncTask<Void?, Void?, Void?>() {
        override fun doInBackground(vararg params: Void?): Void? {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                val name: CharSequence = "MangaChannel"
                val description = "Channel for Manga Reminder"
                val channel = NotificationChannel("manga", name, NotificationManager.IMPORTANCE_DEFAULT)
                channel.description = description
                val notificationManager = weakContext.get()!!.getSystemService(NotificationManager::class.java)
                notificationManager.createNotificationChannel(channel)
            }
            return null
        }
    }
}